internal/dataplane/translator/subtranslator/ingress_atc.go

```diff
@@ -38,7 +35,13 @@ func (m *ingressTranslationMeta) translateIntoKongExpressionRoute() *kongstate.R
 		ingressHost = strings.ReplaceAll(ingressHost, "*", "_")
 	}
 
-	routeName := fmt.Sprintf("%s.%s.%s.%s.%s", m.parentIngress.GetNamespace(), m.parentIngress.GetName(), m.serviceName, ingressHost, m.servicePort.CanonicalString())
+	var routeName string
+	if m.backend.backendType == ingressPathBackendTypeKongServiceFacade {
+		routeName = fmt.Sprintf("%s.%s.%s.%s.svc.facade", m.parentIngress.GetNamespace(), m.parentIngress.GetName(), m.backend.name, ingressHost)
+	} else {
+		routeName = fmt.Sprintf("%s.%s.%s.%s.%s", m.parentIngress.GetNamespace(), m.parentIngress.GetName(), m.backend.name, ingressHost, m.backend.port.CanonicalString())
+	}
```

Comment:
Those naming schemes are repeated verbatim in `internal/dataplane/translator/subtranslator/ingress.go:275`. Maybe introduce a helper(s) to have them defined only in one place

Reply:
Good point, there already is `m.backend.intoKongRouteName` for that. I used it here.